addons/mail/static/src/new/messaging_menu/messaging_menu.js

```diff
@@ -45,4 +45,14 @@ export class MessagingMenu extends Component {
         // without dropdownitem...
         document.body.click();
     }
+
+    getMenuCounter() {
+        const menuCounter =
+            this.messaging.state.discuss.inbox.counter +
+            Object.values(this.messaging.state.threads).filter((thread) => thread.is_pinned)
+                .length +
+            Object.keys(this.messaging.state.notificationGroups).length +
```

Comment:
Note that a group might contribute a lot to counter: e.g. if you have needaction on 10 same group, it contributes to 10 on the counter.